site/content/integrate/apps/functionality/oauth2.md

```diff
@@ -0,0 +1,269 @@
+    E --> F[App exchanges authentication<br/>code for token]
+    F --> G[App stores user's token]
+{{</mermaid>}}
+
+{{<note "App OAuth2 endpoints:">}}
+The `/oauth2/connect` and `/oauth2/complete` calls can be customized using the `get_oauth2_connect_url` and `on_oauth2_complete` parameters of the App's manifest, respectively.
+{{</note>}}
+
+## Use HTTP REST endpoints
+
+### Endpoint URL
+
+OAuth2 data is managed using a single HTTP REST endpoint:
+
+`<mattermost_site_url>/plugins/com.mattermost.apps/api/v1/oauth2`
+
+Replace `<mattermost_site_url>` with the base URL to the Mattermost server.
+The `<mattermost_site_url>` value can be obtained from a [call request context]({{<ref "/integrate/apps/structure/call#context">}}).
+
+#### Authorization
+
+An authorization token is required when invoking HTTP REST endpoints for OAuth2. The token must be set in the `Authorization` header as a bearer token. The `bot_access_token` or `acting_user_access_token` field of the call request context will contain the token; either token can be used.
+
+### Register a remote OAuth2 provider with HTTP REST
+
+Example HTTP request to register the client ID and secret for a remote OAuth2 provider:
+
+```http request
+POST /plugins/com.mattermost.apps/api/v1/oauth2/app HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxx
+Content-Type: application/json
+
+{
+    "client_id": "xxxxxxxxxxxx",
+    "client_secret": "xxxxxxxxxxxx"
+}
+```
+
+{{<note "Updating provider registration:">}}
+To update an existing provider, use the HTTP `PUT` method instead of the HTTP `POST` method.
+{{</note>}}
+
+### Set or update a user access token with HTTP REST
+
+Example HTTP request to set a user access token:
+
+```http request
+POST /plugins/com.mattermost.apps/api/v1/oauth2/user HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxx
+Content-Type: application/json
+
+{
+    "token": "xxxxxxxxxxxx"
+}
+```
+
+{{<note "Updating a user access token:">}}
+To update an existing user access token, use the HTTP `PUT` method instead of the HTTP `POST` method.
+{{</note>}}
+
+### Get a user access token with HTTP REST
+
+Example HTTP request to get a user access token:
+
+```http request
+GET /plugins/com.mattermost.apps/api/v1/oauth2/user HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxx
+Accept: application/json
+```
+
+Example response:
+
+```json
+{
+    "token": "xxxxxxxxxxxx"
+}
+```
+
+## Use a driver
+
+### Register a remote OAuth2 provider with a driver
+
+Example [Golang driver]({{<ref "/integrate/apps/drivers/golang">}}) code to register the OAuth2 provider client ID and client secret:
+
+```go
+// configure is the App call invoked by the system administrator to register OAuth2 provider details.
+// The call should be configured to accept two values: client_id and client_secret.
+func configure(w http.ResponseWriter, req *http.Request) {
+    // Decode the call request data
+	callRequest := new(apps.CallRequest)
```

Comment:
Do the docs ever define/reference the form being used for this submission?

Reply:
No. The example golang code that I took it from does, however. I can link that code here to give further context.

I'd like to avoid including forms in this page since it's primarily a reference on how to use the OAuth2 endpoints/functionality. Forms are described on a different page. That doesn't mean, however, that we can't include a reminder or note about the form needing a particular field or whatnot.